keras_core/optimizers/base_optimizer.py

```diff
@@ -329,6 +329,24 @@ def learning_rate(self, learning_rate):
                 )
             self._learning_rate.assign(learning_rate)
 
+    def set_weights(self, weights):
+        """Set the weights of the optimizer."""
+        if not getattr(self, "_built", False):
```

Comment:
That should be `if not self.built` (I don't think we have a `_built` attribute)

Reply:
Fixed to self.built, thanks!